core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -302,7 +320,13 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
             }
         });
 
-        return mapAcknowledgementFutures(futures);
+        // Update the metrics for the topics for which we have received an acknowledgement.
+        topics.forEach(topic -> {
+            brokerTopicStats.allTopicsStats().totalShareAcknowledgementRequestRate().mark();
```

Comment:
Does this track just the ShareAcknowledgementRequest or the acks in ShareFetchRequest too? If yes, we probably want to name the metric more accurately. If not, this method is called by both ShareFetchRequest and ShareAcknowledgeRequest and we need to separate them out.

Reply:
Yes, I agree with Andrew. That's the intent to track the ShareAcknowlwegment request for topics. Explicit ack and share fetch request we ll track by kafka.network request metrics, as you suggested in other comment.